From 9f6b4b5f066302a0a9808c5389eb17d2c6583471 Mon Sep 17 00:00:00 2001
From: Nathan Chancellor <nathan@kernel.org>
Date: Wed, 4 Oct 2023 11:43:39 -0700
Subject: [PATCH 06/14] WIP: kbuild: Promote
 -Wincompatible-function-pointer-types-strict from W=1

Now that all instances of this have been fixed, move this out of W=1 and
turn it into a hard error, so that potential kCFI failures are caught at
build time, rather than run time.

Signed-off-by: Nathan Chancellor <nathan@kernel.org>
---
 drivers/counter/Makefile   | 2 ++
 scripts/Makefile.extrawarn | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/drivers/counter/Makefile b/drivers/counter/Makefile
index fa3c1d08f706..80e335e836e4 100644
--- a/drivers/counter/Makefile
+++ b/drivers/counter/Makefile
@@ -3,6 +3,8 @@
 # Makefile for Counter devices
 #
 
+ccflags-remove-y := -Werror=incompatible-function-pointer-types-strict
+
 obj-$(CONFIG_COUNTER) += counter.o
 counter-y := counter-core.o counter-sysfs.o counter-chrdev.o
 
diff --git a/scripts/Makefile.extrawarn b/scripts/Makefile.extrawarn
index 174cf7a61df6..fc44dff7864a 100644
--- a/scripts/Makefile.extrawarn
+++ b/scripts/Makefile.extrawarn
@@ -75,6 +75,7 @@ KBUILD_CFLAGS += -Werror=date-time
 
 # enforce correct pointer usage
 KBUILD_CFLAGS += $(call cc-option,-Werror=incompatible-pointer-types)
+KBUILD_CFLAGS += $(call cc-option,-Werror=incompatible-function-pointer-types-strict)
 
 # Require designated initializers for all marked structures
 KBUILD_CFLAGS += $(call cc-option,-Werror=designated-init)
@@ -98,7 +99,6 @@ KBUILD_CFLAGS += $(call cc-option, -Wpacked-not-aligned)
 KBUILD_CFLAGS += $(call cc-option, -Wformat-overflow)
 KBUILD_CFLAGS += $(call cc-option, -Wformat-truncation)
 KBUILD_CFLAGS += $(call cc-option, -Wstringop-truncation)
-KBUILD_CFLAGS += $(call cc-option, -Wincompatible-function-pointer-types-strict)
 
 KBUILD_CPPFLAGS += -Wundef
 KBUILD_CPPFLAGS += -DKBUILD_EXTRA_WARN1
-- 
2.43.0

